components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -3111,4 +3114,41 @@ private Property extractPropertyFromMetadata(MetadataConverter metadataConverter
         }
         return propertyWithName;
     }
+
+    /**
+     * Set the confidential status of federated authenticator properties using metadata.
+     * @param identityProvider Identity Provider.
+     */
+    private void setConfidentialStatusFromMeta(IdentityProvider identityProvider)
+            throws IdentityProviderManagementException {
+
+        Map<String, List<String>> metaFedAuthConfigMap = createFedAuthConfidentialPropsMap();
+        Arrays.asList(identityProvider.getFederatedAuthenticatorConfigs()).forEach(fedAuthConfig -> {
```

Comment:
identityProvider.getFederatedAuthenticatorConfigs() can this be null?

Reply:
IdentityProvider object initializes its federatedAuthenticatorConfigs attribute to an empty array. Also the setter checks for null when setting. So it won't be null.